### PR TITLE
GHA Migration - build misc

### PR DESCRIPTION
**Pull request by mordamax**
Closes https://github.com/paritytech/ci_cd/issues/1017

- Enabled subsystem-benchmarks 100% + adjusted them to publish as separate job so it can use environment for scope secrets
- Added -v1.* triggers for GHA checks
- Removed remove subkey & polkavm from gitlab

### CODE REVIEW

File: .github/workflows/build-misc.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          rustup +nightly show
+
+      - name: Build
+        env:
+          SUBSTRATE_RUNTIME_TARGET: riscv
+        run: |
+          time cargo check -p minimal-template-runtime
+          time cargo check -p westend-runtime
+          time cargo check -p rococo-runtime
+          time cargo check -p polkadot-test-runtime
```
Comment (alvicsam) [inline .github/workflows/build-misc.yml:54]:
```suggestion
          forklift cargo check -p minimal-template-runtime
          forklift cargo check -p westend-runtime
          forklift cargo check -p rococo-runtime
          forklift cargo check -p polkadot-test-runtime
```

File: .github/workflows/build-misc.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        run: |
+          rustup show
+          rustup +nightly show
+
+      - name: Build
+        env:
+          SKIP_WASM_BUILD: 1
+        run: |
+          cd ./substrate/bin/utils/subkey
+          time cargo build --locked --release
```
Comment (alvicsam) [inline .github/workflows/build-misc.yml:75]:
```suggestion
          forklift cargo build --locked --release
```

File: .github/workflows/build-misc.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+on:
```
Comment (alvicsam) [inline .github/workflows/build-misc.yml:1]: Please add the name of the workflow

File: .github/workflows/subsystem-benchmarks.yml [hunk 4]
```diff
@@ -57,26 +51,57 @@ jobs:
           rustup +nightly show
 
       - name: Run Benchmarks
-        continue-on-error: true
         id: run-benchmarks
         run: |
           cargo bench -p ${{ matrix.features.name }} --bench ${{ matrix.features.bench }} --features subsystem-benchmarks || echo "Benchmarks failed"
```
Comment (alvicsam) [inline .github/workflows/subsystem-benchmarks.yml:56]:
```suggestion
          forklift cargo bench -p ${{ matrix.features.name }} --bench ${{ matrix.features.bench }} --features subsystem-benchmarks || echo "Benchmarks failed"
```

**Review by alvicsam — CHANGES_REQUESTED** [review]
Left few comments (forgot to check that forklift is added last time). Also I think the `tags` section should be remove since we use other tags, right @EgorPopelyaev ?

File: .github/workflows/subsystem-benchmarks.yml (hunk 4 above)
Comment (mordamax) [reply .github/workflows/subsystem-benchmarks.yml]: @alvicsam applied all recommendations and updated the system benchmarks publishing to use environment scope + sequenced push to gh-pages 1by1 from artifacts

**Review by EgorPopelyaev — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

File: .github/workflows/check-runtime-migration.yml [hunk 5]
```diff
@@ -46,7 +46,7 @@ jobs:
     # We need to set this to rather long to allow the snapshot to be created, but the average time
     # should be much lower.
     timeout-minutes: 60
-    needs: [set-image]
+    needs: [ set-image ]
```
Comment (alvicsam) [inline .github/workflows/check-runtime-migration.yml:49]: 😑

**Review by alvicsam — CHANGES_REQUESTED** [review]
Nice work, please also add `confirm-all-jobs-passed` to build-misc.yml

**Comment by mordamax** [comment]
> Nice work, please also add `confirm-all-jobs-passed` to build-misc.yml

Thank you, added

File: .github/workflows/build-misc.yml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      - name: Build
+        env:
+          SKIP_WASM_BUILD: 1
+        run: |
+          cd ./substrate/bin/utils/subkey
+          forklift cargo build --locked --release
+
+  confirm-required-build-misc-jobs-passed:
+    runs-on: ubuntu-latest
+    name: All tests passed
```
Comment (alvicsam) [inline .github/workflows/build-misc.yml:90]:
```suggestion
    name: All build misc jobs passed
```
wdyt?

**Review by alvicsam — APPROVED** [review]

**Pull request merged**